pcdsdevices/lxe.py

```diff
@@ -414,6 +414,170 @@ def format_status_info(self, status_info):
+
+    @done.sub_value
+    def _update_position(self, old_value=None, value=None, **kwargs):
+        """The move was completed. Update the notepad readback."""
+        if value == self.done_value and old_value == 0:
+            try:
+                signal = self.notepad_readback
+                position = self.setpoint.get()
+                if signal.connected and signal.write_access:
+                    if signal.get(use_monitor=True) != position:
+                        signal.put(position, wait=False)
+            except Exception as ex:
+                self.log.debug('Failed to update notepad readback to position'
+                               ' %s', position, exc_info=ex)
+
+    @property
+    def limits(self):
+        """
+        Limits as (low_limit, high_limit).
+
+        These Python-only user limits are mirrored from `setpoint`.
+        """
+        return self.setpoint.limits
+
+    @limits.setter
+    def limits(self, limits):
+        # Update the setpoint limits
+        self.setpoint.limits = limits
+
+    def set_current_position(self, position):
+        '''
+        Calculate and configure the user_offset value, indicating the provided
+        ``position`` as the new current position.
+
+        Parameters
+        ----------
+        position
+            The new current position.
+        '''
+        self.user_offset.put(0.0)
+        new_offset = position - self.setpoint.get()
+        self.user_offset.put(new_offset)
+
+    @property
+    def dial_pos(self):
+        """
+        Calculate the dial position.
+
+        The _tgt_time is the actual dial_position in ns.
+
+        Returns
+        -------
+        dial_pos : number
+            The dial position in [s] seconds, or 'N/A' if the dial position is
+            0 or None.
+        """
+        try:
+            dial_pos = self._tgt_time.get()
+            # convert from ns to s
+            return f'{(dial_pos * 1e-9):.3e}'
+        except Exception:
+            return 'N/A'
+
+    def format_status_info(self, status_info):
+        """
+        Override status info handler to render the LXT motor.
+
+        Display lxt motor status info in the ipython terminal.
+
+        Parameters
+        ----------
+        status_info: dict
+            Nested dictionary. Each level has keys name, kind, and is_device.
+            If is_device is True, subdevice dictionaries may follow. Otherwise,
+            the only other key in the dictionary will be value.
+
+        Returns
+        -------
+        status: str
+            Formatted string with all relevant status information.
+        """
+        dial_pos = self.dial_pos
+        position = get_status_float(
+            status_info, 'position', precision=3, format='e')
+        units = get_status_value(status_info, 'setpoint', 'units')
+        return f"""\
+Virtual Motor {self.verbose_name} {self.prefix}
+Current position (user, dial): {position} [{units}] {dial_pos} [s]
+"""
+
+
```

Comment:
I've verified that the methods here are exactly the same as in `LaserTiming`. Is there any consideration for reducing the code duplication? There are probably clean ways to source the common functionality from the same place. Maybe we can refactor in a follow-up PR.

Reply:
Sounds good. As mentioned in a previous comment, I'd like to remove some of the additional functionality that I think will be taken up by the IOC in a future PR. I can refactor the remaining common at the same time.